// react-checkbox-dropdown/es/components/AddValue/index.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Input from './Input';

var AddValue = function AddValue(_ref) {
  var checkAddedValue = _ref.checkAddedValue,
      addIcon = _ref.addIcon,
      onAddValue = _ref.onAddValue,
      inputPlaceholder = _ref.inputPlaceholder,
      styleFunc = _ref.styleFunc,
      inputStyles = _ref.inputStyles,
      buttonStyles = _ref.buttonStyles;

  var _useState = useState(''),
      value = _useState[0],
      setValue = _useState[1];

  var _useState2 = useState(false),
      showInput = _useState2[0],
      setShowInput = _useState2[1];

  var _useState3 = useState(null),
      isSuccess = _useState3[0],
      setSuccess = _useState3[1];

  var onChange = function onChange(_ref2) {
    var target = _ref2.target;

    setValue(target.value);
  };

  var handleShowInput = function handleShowInput() {
    if (!showInput) {
      setShowInput(true);
    }
  };

  var handleAddValue = function handleAddValue() {
    // add if value accepted and ignore if not
    if (checkAddedValue(value)) {
      onAddValue(value);
      setSuccess(true);
      setValue('');
      setShowInput(false);
    } else {
      setSuccess(false);
    }
  };

  var handleClickAddButton = function handleClickAddButton() {
    if (showInput) {
      // ignore if there is no value
      if (!value) {
        setSuccess(false);
        return;
      }
      handleAddValue();
    }
  };
  return React.createElement(
    'div',
    {
      className: 'add-more',
      onClick: handleShowInput,
      style: _extends({
        display: 'flex',
        alignItems: 'center',
        height: '65px',
        paddingLeft: '20px',
        fontSize: '18px',
        color: '#989898'
      }, styleFunc({ isSuccess: isSuccess, isInputShown: showInput }))
    },
    React.createElement(
      'button',
      {
        className: 'add-icon',
        onClick: handleClickAddButton,
        style: _extends({
          height: '20px',
          width: '20px',
          backgroundColor: 'inherit',
          marginRight: '20px',
          cursor: 'pointer',
          padding: '0',
          border: 'none',
          boxShadow: 'none',
          outline: 'none'
        }, buttonStyles({ isSuccess: isSuccess, isInputShown: showInput }))
      },
      addIcon
    ),
    React.createElement(Input, {
      value: value,
      onChange: onChange,
      showInput: showInput,
      isSuccess: isSuccess,
      onSubmit: handleAddValue,
      placeholder: inputPlaceholder,
      styleFunc: inputStyles
    })
  );
};

AddValue.propTypes = process.env.NODE_ENV !== "production" ? {
  checkAddedValue: PropTypes.func.isRequired,
  addIcon: PropTypes.oneOf(PropTypes.element, PropTypes.node),
  onAddValue: PropTypes.func.isRequired,
  inputPlaceholder: PropTypes.string.isRequired,
  styleFunc: PropTypes.func.isRequired,
  inputStyles: PropTypes.func.isRequired,
  buttonStyles: PropTypes.func.isRequired
} : {};

export default AddValue;
