// react-checkbox-dropdown/lib/components/Checkbox/index.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Checkbox = function Checkbox(_ref) {
  var isChecked = _ref.isChecked,
      checkboxStyle = _ref.checkboxStyle,
      checkedStyle = _ref.checkedStyle;

  var _useState = (0, _react.useState)(false),
      isHovered = _useState[0],
      setHover = _useState[1];

  return _react2.default.createElement(
    'div',
    {
      className: 'checkbox',
      onMouseEnter: function onMouseEnter() {
        setHover(true);
      },
      onMouseLeave: function onMouseLeave() {
        setHover(false);
      },
      style: _extends({
        height: '20px',
        width: '20px',
        border: '1px solid #000',
        marginRight: '20px',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center'
      }, checkboxStyle({ isChecked: isChecked, isHovered: isHovered }))
    },
    isChecked && _react2.default.createElement('div', {
      className: 'checked',
      style: _extends({
        width: '60%',
        height: '60%',
        backgroundColor: '#A39A9A'
      }, checkedStyle())
    })
  );
};

exports.default = Checkbox;
module.exports = exports['default'];
