// react-checkbox-dropdown/lib/components/Options/index.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Options = function Options(_ref) {
  var isOpen = _ref.isOpen,
      children = _ref.children,
      styleFunc = _ref.styleFunc;

  return _react2.default.createElement(
    'div',
    {
      className: 'options',
      style: _extends({
        transform: 'scaleY(' + (isOpen ? '1' : '0') + ')',
        height: isOpen ? 'auto' : '0',
        maxHeight: isOpen ? '650px' : '0',
        overflowY: 'scroll',
        transition: 'all .3s ease-in-out'
      }, styleFunc({ isOpen: isOpen }))
    },
    children
  );
};

Options.propTypes = process.env.NODE_ENV !== "production" ? {
  isOpen: _propTypes2.default.bool.isRequired,
  children: _propTypes2.default.arrayOf([_propTypes2.default.node, _propTypes2.default.element]).isRequired,
  style: _propTypes2.default.object
} : {};

Options.defaultProps = {
  styleFunc: function styleFunc() {}
};

exports.default = Options;
module.exports = exports['default'];
