// react-checkbox-dropdown/lib/components/Tag/index.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Tag = function Tag(_ref) {
  var option = _ref.option,
      onDeselect = _ref.onDeselect,
      icon = _ref.icon,
      styleFunc = _ref.styleFunc;

  var _useState = (0, _react.useState)(false),
      isHovered = _useState[0],
      setHover = _useState[1];

  return _react2.default.createElement(
    'div',
    {
      className: 'tag',
      style: _extends({
        display: 'flex',
        alignContent: 'center',
        margin: '0 20px 10px 0',
        borderRadius: '5px',
        backgroundColor: '#C4C4C4',
        color: '#2D2626',
        fontSize: '12px',
        padding: '4px'
      }, styleFunc({ isHovered: isHovered })),
      onMouseEnter: function onMouseEnter() {
        setHover(true);
      },
      onMouseLeave: function onMouseLeave() {
        setHover(false);
      }
    },
    _react2.default.createElement(
      'p',
      {
        style: {
          margin: 0,
          marginRight: '8px'
        }
      },
      option.label
    ),
    _react2.default.createElement(
      'div',
      {
        className: 'tag-close-icon',
        onClick: function onClick() {
          onDeselect(option);
        },
        style: { height: 'auto', cursor: 'pointer' }
      },
      icon
    )
  );
};

exports.default = Tag;
module.exports = exports['default'];
