// react-checkbox-dropdown/lib/components/Tags/index.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Tags = function Tags(_ref) {
  var children = _ref.children,
      styleFunc = _ref.styleFunc;

  if (!children.length) return null;
  return _react2.default.createElement(
    'div',
    {
      className: 'tags',
      style: _extends({
        display: 'flex',
        flexWrap: 'wrap',
        padding: '10px 0',
        paddingLeft: '22px'
      }, styleFunc())
    },
    children
  );
};

Tags.propTypes = process.env.NODE_ENV !== "production" ? {
  children: _propTypes2.default.arrayOf(_propTypes2.default.element).isRequired,
  styleFunc: _propTypes2.default.func.isRequired
} : {};

exports.default = Tags;
module.exports = exports['default'];
