// react-checkbox-dropdown/lib/components/AddValue/Input.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Input = function Input(_ref) {
  var value = _ref.value,
      onChange = _ref.onChange,
      showInput = _ref.showInput,
      placeholder = _ref.placeholder,
      isSuccess = _ref.isSuccess,
      onSubmit = _ref.onSubmit,
      styleFunc = _ref.styleFunc;

  var _useState = (0, _react.useState)(false),
      isFocused = _useState[0],
      setFocus = _useState[1];

  var handleKeyDown = function handleKeyDown(event) {
    if (isFocused) {
      console.log(event.keyCode);
      if (event.keyCode === 13) {
        onSubmit();
      }
    }
  };

  return showInput ? _react2.default.createElement('input', {
    type: 'text',
    value: value,
    onChange: onChange,
    onKeyDown: handleKeyDown,
    onFocus: function onFocus() {
      setFocus(true);
    },
    onBlur: function onBlur() {
      setFocus(false);
    },
    style: _extends({
      height: '50px',
      width: '100%',
      border: '1px solid ' + (!isSuccess ? '#dc3545' : '#000'),
      borderRadius: '12px',
      marginRight: '20px',
      paddingLeft: '10px',
      fontSize: '18px'
    }, styleFunc({ isFocused: isFocused, isSuccess: isSuccess })),
    placeholder: !isSuccess ? 'Please enter a valid option' : ''
  }) : _react2.default.createElement(
    'p',
    { className: 'placeholder' },
    placeholder
  );
};

Input.propTypes = process.env.NODE_ENV !== "production" ? {
  value: _propTypes2.default.string.isRequired,
  onChange: _propTypes2.default.func.isRequired,
  showInput: _propTypes2.default.bool.isRequired,
  placeholder: _propTypes2.default.string.isRequired,
  isSuccess: _propTypes2.default.bool.isRequired,
  onSubmit: _propTypes2.default.func.isRequired,
  styleFunc: _propTypes2.default.func.isRequired
} : {};

exports.default = Input;
module.exports = exports['default'];
