// react-checkbox-dropdown/lib/index.js
'use strict';

exports.__esModule = true;
exports.createStyles = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Options = require('./components/Options');

var _Options2 = _interopRequireDefault(_Options);

var _Option = require('./components/Option');

var _Option2 = _interopRequireDefault(_Option);

var _Checkbox = require('./components/Checkbox');

var _Checkbox2 = _interopRequireDefault(_Checkbox);

var _Tags = require('./components/Tags');

var _Tags2 = _interopRequireDefault(_Tags);

var _Tag = require('./components/Tag');

var _Tag2 = _interopRequireDefault(_Tag);

var _AddValue = require('./components/AddValue');

var _AddValue2 = _interopRequireDefault(_AddValue);

var _initOptions = require('./helpers/initOptions');

var _initOptions2 = _interopRequireDefault(_initOptions);

var _checkSelectedOption = require('./helpers/checkSelectedOption');

var _checkSelectedOption2 = _interopRequireDefault(_checkSelectedOption);

var _icons = require('./icons');

var _createStyles = require('./createStyles');

var _createStyles2 = _interopRequireDefault(_createStyles);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function CheckboxDropdownComponent(_ref) {
  var options = _ref.options,
      value = _ref.value,
      displayText = _ref.displayText,
      onChange = _ref.onChange,
      onOpen = _ref.onOpen,
      onClose = _ref.onClose,
      checkAddedValue = _ref.checkAddedValue,
      style = _ref.style,
      openIcon = _ref.openIcon,
      closeIcon = _ref.closeIcon,
      tagXIcon = _ref.tagXIcon,
      addIcon = _ref.addIcon,
      isStrict = _ref.isStrict,
      onDeselectOption = _ref.onDeselectOption,
      displayTags = _ref.displayTags,
      inputPlaceholder = _ref.inputPlaceholder;

  var _useState = (0, _react.useState)(false),
      isOpen = _useState[0],
      setOpen = _useState[1];

  var _useState2 = (0, _react.useState)(false),
      isFocused = _useState2[0],
      setFocus = _useState2[1];

  var containerRef = (0, _react.useRef)(false);

  var container = style.container,
      optionStyles = style.option,
      optionsStyles = style.options,
      toggleButton = style.toggleButton,
      checkboxBody = style.checkboxBody,
      checkboxDot = style.checkboxDot,
      addValue = style.addValue,
      addValueButton = style.addValueButton,
      addValueInput = style.addValueInput,
      displayTextStyles = style.displayText,
      tagsContainer = style.tagsContainer,
      tag = style.tag;


  (0, _react.useEffect)(function () {
    // close the dropdown when clicking away
    function onClickAway(event) {
      var node = containerRef.current;
      if (node && !node.contains(event.target)) {
        setOpen(false);
      }
    }

    if (isOpen && containerRef) {
      document.addEventListener('click', onClickAway);
    } else {
      document.removeEventListener('click', onClickAway);
    }
  }, [isOpen, containerRef]);

  (0, _react.useEffect)(function () {
    if (isOpen) {
      onOpen();
    } else {
      onClose();
    }
  }, [isOpen]);

  var onCloseTag = function onCloseTag(option) {
    onChange(option);
    onDeselectOption(option);
  };

  var onAddAnother = function onAddAnother(value) {
    var newOption = { label: value, value: value };
    onChange(newOption);
  };

  return _react2.default.createElement(
    'div',
    {
      className: 'container',
      ref: containerRef,
      style: _extends({
        width: '100%',
        maxWidth: '570px',
        border: '.5px solid ' + (isFocused ? '#377caf' : '#797474'),
        overflow: 'hidden',
        boxShadow: '0px 4px 4px rgba(0, 0, 0, 0.25)',
        borderRadius: '5px',
        backgroundColor: '#fff',
        fontFamily: 'SFMono-Regular,Menlo,Monaco,Consolas,"Liberation Mono","Courier New",monospace'
      }, container({ isOpen: isOpen, isFocused: isFocused }))
    },
    _react2.default.createElement(
      'button',
      {
        className: 'display-text-button',
        style: _extends({
          minHeight: '60px',
          display: 'flex',
          alignItems: 'center',
          cursor: 'pointer',
          width: '100%',
          boxShadow: 'none',
          backgroundColor: 'inherit',
          borderRadius: '5px',
          border: '.5px solid transparent',
          padding: '0',
          margin: '0'
        }, toggleButton({ isOpen: isOpen, isFocused: isFocused })),
        onClick: function onClick() {
          setOpen(!isOpen);
        },
        onFocus: function onFocus() {
          setFocus(true);
        },
        onBlur: function onBlur() {
          setFocus(false);
        }
      },
      _react2.default.createElement(
        'h3',
        {
          className: 'display-text',
          style: _extends({
            paddingLeft: '22px',
            fontSize: '18px',
            marginRight: 'auto'
          }, displayTextStyles({ isOpen: isOpen, isFocused: isFocused }))
        },
        displayText
      ),
      _react2.default.createElement(
        'div',
        {
          className: 'icon',
          style: {
            height: '24px',
            width: '24px',
            marginRight: '10px',
            marginLeft: '10px',
            color: 'red'
          }
        },
        isOpen ? openIcon : closeIcon
      )
    ),
    displayTags && _react2.default.createElement(
      _Tags2.default,
      { styleFunc: tagsContainer },
      value.map(function (option) {
        return _react2.default.createElement(_Tag2.default, {
          key: 'tag-' + option.value,
          option: option,
          icon: tagXIcon,
          onDeselect: onCloseTag,
          styleFunc: tag
        });
      })
    ),
    _react2.default.createElement(
      _Options2.default,
      { isOpen: isOpen, styleFunc: optionsStyles },
      (0, _initOptions2.default)(options).map(function (option, index) {
        return _react2.default.createElement(_Option2.default, {
          key: 'option-' + option.value,
          index: index,
          option: option,
          onChange: onChange,
          onDeselectOption: onDeselectOption,
          Checkbox: _react2.default.createElement(_Checkbox2.default, {
            isChecked: (0, _checkSelectedOption2.default)(option, value),
            checkboxStyle: checkboxBody,
            checkedStyle: checkboxDot
          }),
          isSelected: (0, _checkSelectedOption2.default)(option, value),
          styleFunc: optionStyles
        });
      }),
      !isStrict && _react2.default.createElement(_AddValue2.default, {
        onAddValue: onAddAnother,
        addIcon: addIcon,
        checkAddedValue: checkAddedValue,
        inputPlaceholder: inputPlaceholder,
        styleFunc: addValue,
        inputStyles: addValueInput,
        buttonStyles: addValueButton
      })
    )
  );
}

CheckboxDropdownComponent.propTypes = process.env.NODE_ENV !== "production" ? {
  options: _propTypes2.default.arrayOf(_propTypes2.default.objectOf({
    label: _propTypes2.default.string,
    value: _propTypes2.default.any
  })).isRequired,
  value: _propTypes2.default.arrayOf(_propTypes2.default.objectOf({
    label: _propTypes2.default.string,
    value: _propTypes2.default.string
  })).isRequired,
  displayText: _propTypes2.default.string.isRequired,
  onChange: _propTypes2.default.func.isRequired,
  onOpen: _propTypes2.default.func,
  onClose: _propTypes2.default.func,
  checkAddedValue: _propTypes2.default.func,
  style: _propTypes2.default.objectOf({
    container: _propTypes2.default.func,
    option: _propTypes2.default.func,
    options: _propTypes2.default.func,
    toggleButton: _propTypes2.default.func,
    displayText: _propTypes2.default.func,
    checkboxBody: _propTypes2.default.func,
    checkboxDot: _propTypes2.default.func,
    addValue: _propTypes2.default.func,
    addValueButton: _propTypes2.default.func,
    addValueInput: _propTypes2.default.func,
    tagsContainer: _propTypes2.default.func,
    tag: _propTypes2.default.func
  }),
  openIcon: _propTypes2.default.oneOfType([_propTypes2.default.node, _propTypes2.default.element]),
  closeIcon: _propTypes2.default.oneOfType([_propTypes2.default.node, _propTypes2.default.element]),
  tagXIcon: _propTypes2.default.oneOfType([_propTypes2.default.node, _propTypes2.default.element]),
  addIcon: _propTypes2.default.oneOfType([_propTypes2.default.node, _propTypes2.default.element]),
  isStrict: _propTypes2.default.bool,
  onDeselectOption: _propTypes2.default.func,
  displayTags: _propTypes2.default.bool,
  inputPlaceholder: _propTypes2.default.string
} : {};

CheckboxDropdownComponent.defaultProps = {
  onOpen: function onOpen() {},
  onClose: function onClose() {},
  checkAddedValue: function checkAddedValue() {
    return true;
  },

  style: {
    container: function container() {
      return {};
    },
    option: function option() {
      return {};
    },
    options: function options() {
      return {};
    },
    toggleButton: function toggleButton() {
      return {};
    },
    displayText: function displayText() {
      return {};
    },
    checkboxBody: function checkboxBody() {
      return {};
    },
    checkboxDot: function checkboxDot() {
      return {};
    },
    addValue: function addValue() {
      return {};
    },
    addValueButton: function addValueButton() {
      return {};
    },
    addValueInput: function addValueInput() {
      return {};
    },
    tagsContainer: function tagsContainer() {
      return {};
    },
    tag: function tag() {
      return {};
    }
  },
  openIcon: _react2.default.createElement(_icons.UpIcon, null),
  closeIcon: _react2.default.createElement(_icons.DownIcon, null),
  tagXIcon: _react2.default.createElement(_icons.XIcon, null),
  addIcon: _react2.default.createElement(_icons.AddIcon, null),
  isStrict: true,
  onDeselectOption: function onDeselectOption() {},

  displayTags: false,
  inputPlaceholder: 'Add Another Value'
};

exports.createStyles = _createStyles2.default;
exports.default = CheckboxDropdownComponent;
