// react-checkbox-dropdown/es/components/Option/index.js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

import React, { useState } from 'react';
import PropTypes from 'prop-types';

var Option = function Option(_ref) {
  var _onChange = _ref.onChange,
      onDeselectOption = _ref.onDeselectOption,
      option = _ref.option,
      isSelected = _ref.isSelected,
      styleFunc = _ref.styleFunc,
      index = _ref.index,
      Checkbox = _ref.Checkbox;

  var _useState = useState(false),
      isFocused = _useState[0],
      setFocus = _useState[1];

  var _useState2 = useState(false),
      isHovered = _useState2[0],
      setHover = _useState2[1];

  return React.createElement(
    'label',
    {
      htmlFor: 'option-' + index,
      className: 'option',
      onMouseEnter: function onMouseEnter() {
        setHover(true);
      },
      onMouseLeave: function onMouseLeave() {
        setHover(false);
      },
      onFocus: function onFocus() {
        setFocus(true);
      },
      onBlur: function onBlur() {
        setFocus(false);
      },
      style: _extends({
        fontSize: '18px',
        display: 'flex',
        alignItems: 'center',
        paddingLeft: '22px',
        height: '50px',
        borderBottom: '.5px solid #A39A9A',
        fontWeight: isSelected ? 'bold' : 'normal',
        cursor: 'pointer'
      }, styleFunc({ isSelected: isSelected, isFocused: isFocused, isHovered: isHovered }))
    },
    React.createElement('input', {
      id: 'option-' + index,
      type: 'checkbox',
      style: { display: 'none' },
      value: isSelected,
      checked: isSelected,
      onChange: function onChange() {
        _onChange(option);
        if (isSelected) {
          onDeselectOption(option);
        }
      }
    }),
    Checkbox,
    React.createElement(
      'p',
      null,
      option.label
    )
  );
};

Option.propTypes = process.env.NODE_ENV !== "production" ? {
  onChange: PropTypes.func.isRequired,
  onDeselectOption: PropTypes.func,
  option: PropTypes.shape({
    key: PropTypes.any,
    label: PropTypes.string,
    value: PropTypes.string
  }).isRequired,
  isSelected: PropTypes.bool.isRequired,
  styleFunc: PropTypes.func,
  index: PropTypes.number.isRequired,
  Checkbox: PropTypes.element.isRequired
} : {};

Option.defaultProps = {
  onDeselectOption: function onDeselectOption() {},
  styleFunc: function styleFunc() {}
};

export default Option;
